.openshift-ci/get_latest_helm_chart_versions.py

```diff
@@ -84,6 +96,83 @@ def get_latest_helm_chart_version_for_specific_release(chart_name, release):
         remove_helm_repo()
 
 
+def get_supported_helm_chart_versions():
+    add_helm_repo()
+    try:
+        update_helm_repo()
+        return __get_supported_helm_chart_versions()
+    finally:
+        remove_helm_repo()
+
+
+def __get_supported_helm_chart_versions():
+    supported_central_versions = []
+    supported_sensor_versions = []
+
+    supported_releases = get_supported_releases()
+    for release in supported_releases:
+        if __does_chart_exist("stackrox-central-services", release):
+            supported_central_versions.append(__get_latest_helm_chart_version_for_specific_release(
+                "stackrox-central-services", release)
+            )
+        else:
+            logging.debug(f"Supported version \"{release.major}.{release.minor}\" has no corresponding helm chart for "
+                          f"stackrox-central-services.")
+        if __does_chart_exist("stackrox-secured-cluster-services", release):
+            supported_sensor_versions.append(__get_latest_helm_chart_version_for_specific_release(
+                "stackrox-secured-cluster-services", release)
+            )
+        else:
+            logging.debug(f"Supported version \"{release.major}.{release.minor}\" has no corresponding helm chart for "
+                          f"stackrox-secured-cluster-services.")
+    return supported_central_versions, supported_sensor_versions
+
+
+def get_supported_releases():
+    supported_releases = []
+    data = __get_data_from_product_lifecycles_api()
+    releases = []
+    try:
+        releases = data["data"][0]["versions"]
+    except Exception as exception:
+        logging.debug(f"Found no RHACS releases in PRODUCT_LIFECYCLES_API at "
+                      f"{PRODUCT_LIFECYCLES_API}\n{repr(exception)}")
```

Comment:
If you want to go the python way and use the `try-catch`, I would normally catch the specific exceptions this piece of code can throw; in this case `KeyError` and `IndexError`.

You can also check if a particular key is in a dictionary with `"key" in data`. Similarly with the length of the array you can use `len(array) > 0`.With this you can avoid the `try-catch`.

Reply:
This is one thing I dislike about python, there is no good way to know which exceptions a library function may throw without reading all the code. I've looked at the documentation for the functions I'm using and am trying to narrow it down now, but of course this is no _guarantee_ for safety.